### PR TITLE
fix: Update onboarding form and update documentation for bitbucket

### PR DESCRIPTION
**Pull request by rephus**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes:  #1783

### Description:
Update onboarding form and update documentation for bitbucket

### Preview:

Peek 2022-05-03 09-26

Screenshot from 2022-05-03 09-35-32

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [x] Docs  
- [ ] CI/CD

### CODE REVIEW

File: packages/frontend/src/components/ProjectConnection/DbtForms/BitBucketForm.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ const BitBucketForm: FC<{ disabled: boolean }> = ({ disabled }) => {
             <PasswordInput
                 name="dbt.personal_access_token"
                 label="Personal access token"
```
Comment (ZeRego) [inline packages/frontend/src/components/ProjectConnection/DbtForms/BitBucketForm.tsx:33]:
```suggestion
                label="HTTP access token"
```

File: packages/frontend/src/components/ProjectConnection/DbtSettingsForm.tsx [hunk 2]
```diff
@@ -53,6 +55,8 @@ const DbtSettingsForm: FC<DbtSettingsFormProps> = ({
     }, [health, type]);
 
     const form = useMemo(() => {
+        reset({ 'dbt.type': type }); // Reset all defaultValues in forms
```
Comment (ZeRego) [inline packages/frontend/src/components/ProjectConnection/DbtSettingsForm.tsx:58]: Is this working well when editing an existing project ?

Comment (rephus) [reply packages/frontend/src/components/ProjectConnection/DbtSettingsForm.tsx:58]: It looks good to me 

Peek 2022-05-03 10-24

**Review by TuringLovesDeathMetal — APPROVED** [review]
niiiice!

The `http access token` isn't linking to anything in the docs in the PR environment when you go to the `connect a project` form. But I imagine that will change once the docs changes are merged too?

**Pull request merged**